src/runtime/runtime.ts

```diff
@@ -1140,8 +1190,12 @@ export class Transaction {
       index.insert(change);
       changeIx++;
     }
-  }
 
+    // Once the transaction is effectively done, we need to clean up after ourselves. We
+    // arena allocated a bunch of IDs related to function call outputs, which we can now
+    // safely release.
+    GlobalInterner.releaseArena("functionOutput");
```

Comment:
the only problem is that we've have to pass that information down through everything somehow that isn't insane. Since we're pretty far off from having multiple threads (and maybe never will?). I think we can worry about that later.